src/App.js

```diff
@@ -1,15 +1,30 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
 import TodoList from './TodoList'
 import AddTodoForm from './AddTodoForm'
 
+function useSemiPersistentState(key, initialState) {
+	const savedValue = localStorage.getItem(key) || initialState
+	console.log(savedValue)
+
+	const [value, setValue] = useState(JSON.parse(savedValue))
+
+	useEffect(() => {
+		localStorage.setItem(key, JSON.stringify(value))
+	}, [key, value])
+	return [value, setValue]
+}
 function App() {
-	const [newTodo, setNewTodo] = React.useState('')
+	const [todoList, setTodoList] = useSemiPersistentState('savedTodoList', [])
```

Comment:
Nice demonstration of using your custom hook! I don't believe there is explicit direction to do this but good job!

Remember you want your custom hook to be abstract so that you can reuse it. If you were to hardcode these values in your custom hook that would limit its modularity.